openshift/kustomize/tekton/base/tasks/deploy-all.yaml

```diff
@@ -0,0 +1,122 @@
+kind: Task
+metadata:
+  name: deploy-all
+  annotations:
+    tekton.dev/displayName: openshift client
+    tekton.dev/pipelines.minVersion: 0.12.1
+    tekton.dev/tags: cli
+    description: >-
+      This task will tag a specific image with the environment name.
+      This will deploy the image if there is a DeployConfig trigger listening for the environment tag.
+  labels:
+    name: oc-all
+    part-of: tno
+    version: 1.0.0
+    component: task
+    managed-by: kustomize
+spec:
+  params:
+    - name: PROJECT_SHORTNAME
+      description: The project namespace.
+      type: string
+      default: 9b301c
+    - name: IMAGE_TAG
+      description: The tag given to the built image.
+      type: string
+      default: latest
+    - name: ENV
+      description: The environment to deploy to.
+      type: string
+    - name: TIMEOUT
+      description: The timeout before it will stop waiting for the pod to become available.
+      type: string
+      default: "600s"
+  workspaces:
+    - name: conditions
+      mountPath: /data
+
+  steps:
+    - name: deploy
+      image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+      resources: {}
+      script: |
+        set -xe
+
+        # Place config into environment variables.
+        if test -f $(workspaces.conditions.path)/build.env; then
+          export $(grep -v '^#' $(workspaces.conditions.path)/build.env | xargs)
+        else
+          echo 'Workspace conditions build.env not found'
+          exit 1;
+        fi
+
+        # An array of all the objects for the solution.
+        SERVICES=("content" "filecopy" "filemonitor" "image" "indexing" "nlp" "notification" "reporting" "syndication" "transcription" "capture" "clip")
+        COMPONENTS=("api" "editor" "subscriber" "charts-api" "content" "filecopy" "filemonitor" "image" "indexing" "nlp" "notification" "reporting" "syndication" "transcription" "capture" "clip")
+        declare -A IMAGES=(["api"]="api" ["editor"]="editor" ["subscriber"]="subscriber" ["charts-api"]="charts-api" ["content"]="content-service" ["filecopy"]="filecopy-service" ["filemonitor"]="filemonitor-service" ["image"]="image-service" ["indexing"]="indexing-service" ["nlp"]="nlp-service" ["notification"]="notification-service" ["reporting"]="reporting-service" ["syndication"]="syndication-service" ["transcription"]="transcription-service" ["capture"]="capture-service" ["clip"]="clip-service")
+        declare -A OBJECTS=(["api"]="sts" ["editor"]="dc" ["subscriber"]="dc" ["charts-api"]="dc" ["content"]="dc" ["filecopy"]="dc" ["filemonitor"]="dc" ["image"]="dc" ["indexing"]="dc" ["nlp"]="dc" ["notification"]="dc" ["reporting"]="dc" ["syndication"]="dc" ["transcription"]="dc" ["capture"]="dc" ["clip"]="dc")
+        declare -A REPLICAS=(["content"]="1" ["filecopy"]="1" ["filemonitor"]="1" ["image"]="1" ["indexing"]="1" ["nlp"]="1" ["notification"]="1" ["reporting"]="1" ["syndication"]="1" ["transcription"]="1" ["capture"]="0" ["clip"]="0")
```

Comment:
Annoyingly we must keep a static configuration of how many replicas we want to spin up.